### PR TITLE
xfce: cleanup, hyphenate attributes

### PR DESCRIPTION
**Pull request by lukateras**
###### Motivation for this change

This is groundwork for adding Xfce 4.13.

I want to provide Xfce attrset as a NixOS option in https://github.com/NixOS/nixpkgs/pull/32763, and to do that, I will need to have consistent attribute names between Xfce versions.

I've rebuilt system with this change.

/cc @vcunat @volth

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/xfce/default.nix [hunk 1]
```diff
@@ -1,106 +1,208 @@
...
+
+  #### GTK+3 (deprecated, see NixOS/nixpkgs#32763)
+
+  libxfce4ui_gtk3 = libxfce4ui.override { withGtk3 = true; };
+
+  xfce4panel_gtk3 = xfce4-panel.override { withGtk3 = true; };
+
+  xfce4_power_manager_gtk3 = xfce4-power-manager.override { withGtk3 = true; };
+
+  #### ALIASES
```
Comment (volth) [inline pkgs/desktops/xfce/default.nix]: Should these aliases be maintained?
nixpkgs-unstable is allowed to break the thinks, isn't it?

Comment (yegortimoshenko) [reply pkgs/desktops/xfce/default.nix]: This would be some drastic breakage, I'd rather wait until Xfce 4.12 is phased out.

**Review by vcunat — APPROVED** [review]
Seems good.  (Though hard to see how all the code moves.)

**Comment by volth** [comment]
Maybe old codes could print warnings when used?

```
xfce4_timer_plugin = builtins.trace "'xfce4_timer_plugin' is deprecated" xfce4-timer-plugin;
```

**Comment by vcunat** [comment]
There have been discussions about that.  So far, noone managed a way that wouldn't be annoying with `nix-env -qa`, as you can't really make a difference...

**Comment by volth** [comment]
BTW, this renaming-with-aliases  is a breaking change as well. It breaks things silently making them hard to debug (that is why I initially proposed not to keep old names at all):

if there is an overlay making something like (in my nixpkgs I have at least 4 such overrides)
```
(self: super: {
  xfce4-12 = super.xfce4-12 // {
    xfce4_namebar_plugin = super.xfce4-12.xfce4_namebar_plugin.overrideDerivation(old: {
      # config-xfce4/panel/namebar-3.rc stores absolute path to theme including /nix/store, which could 
be garbage collected
      postPatch = old.postPatch + ''
        substituteInPlace src/namebar.vala --replace "namebar/themes/Default" 
"namebar/themes/Ambiance"
      '';
    });
  };
})
```

```xfce4_namebar_plugin``` must be changed to ```xfce4-namebar-plugin``` here in order ```.overrideDerivation``` keep working.

But it is not clear which names to change, because if it is not renamed, there will be no error nor warning, just a wrong attribute will be overriden - the one which are not used since the rename.

I'd prefer this overlay code to fail with "no such name". Or at least to warn about using a deprecated name.

**Comment by vcunat** [comment]
Hmm, then we're doing either aliases or overlays wrong :-)  but maybe it's a bit difficult.

**Comment by volth** [comment]
It resembles co/contra-variance more than wrong/right.
aliases works well in ```nix-env``` or ```environment.systemPackages``` but when in comes to overrides, only one of aliases does work, and it is not easy to find a problem caused by changing of that main alias.

**Comment by vcunat** [comment]
By "wrong" I meant that my expectation from aliases is that whatever I do, it doesn't matter which name I use (except for the deprecation warning, perhaps).

**Pull request merged**